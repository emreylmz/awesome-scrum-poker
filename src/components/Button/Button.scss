@import "../../styles/colors";
@import "../../styles/metrics";

.button {
  display: flex;
  font-size: 18px;
  padding: $default-margin;
  background-color: aliceblue;
  color: $primary-color;
  border-radius: $default-border-radius;
  flex: 0;
  text-align: center;
  margin: $default-margin;
  border: $default-border-width solid $primary-color;
  cursor: pointer;

  &:hover {
    background-color: $secondary-color;
  }
}
