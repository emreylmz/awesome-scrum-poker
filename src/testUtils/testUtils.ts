export const emptyTestFunc = (): void => void 0
