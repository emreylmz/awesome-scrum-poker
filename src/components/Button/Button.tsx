import React from 'react'
import { Render } from '../../types'
import { classNames } from '../../utils'
import './Button.scss'

interface Props {
  className?: string
  title: string
  disabled?: boolean
  onClick: () => void
}

class Button extends React.PureComponent<Props> {
  render(): Render {
    const { className, title, disabled, onClick } = this.props

    const buttonClassName = classNames('button', disabled && 'disabled', className)
    return (
      <div className={buttonClassName} onClick={onClick}>
        <span className={'button__title'}>{title}</span>
      </div>
    )
  }
}

export type ButtonProps = Props
export default Button
