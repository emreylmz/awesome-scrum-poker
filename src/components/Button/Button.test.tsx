import { shallow, ShallowWrapper } from 'enzyme'
import React from 'react'
import sinon from 'sinon'
import '../../testUtils/setupTests'
import { emptyTestFunc } from '../../testUtils/testUtils'
import Button, { ButtonProps } from './Button'

type DoneCallback = jest.DoneCallback

const defaultTitle = 'My Button'

const getWrapper = (props?: ButtonProps): ShallowWrapper => {
  return shallow(<Button title={defaultTitle} onClick={emptyTestFunc} {...props} />)
}

describe('Button UI', (): void => {
  it('should render correctly', (done: DoneCallback): void => {
    const wrapper = getWrapper()
    expect(wrapper).toMatchSnapshot()
    done()
  })

  it('should have title', (done: DoneCallback): void => {
    const title = 'Test Button'
    const wrapper = getWrapper({ title, onClick: emptyTestFunc })
    const titleWrapper = wrapper.find('.button__title')
    expect(titleWrapper.text()).toEqual(title)
    done()
  })

  it('should set title with props', (done: DoneCallback): void => {
    const title = 'Test Button'
    const wrapper = getWrapper()
    let titleWrapper = wrapper.find('.button__title')
    expect(titleWrapper.text()).toEqual(defaultTitle)
    wrapper.setProps({ title })
    titleWrapper = wrapper.find('.button__title')
    expect(titleWrapper.text()).toEqual(title)
    done()
  })

  it('should be able to disabled with props', (done: DoneCallback): void => {
    const wrapper = getWrapper()
    let voteComponent = wrapper.find('.button')
    expect(voteComponent.hasClass('disabled')).toBeFalsy()
    wrapper.setProps({ disabled: true })
    voteComponent = wrapper.find('.button')
    expect(voteComponent.hasClass('disabled')).toBeTruthy()
    done()
  })
})

describe('Button Logic', () => {
  it('should call onClick method when clicked', (done: DoneCallback): void => {
    const testFunc = sinon.spy()
    const wrapper = getWrapper({ title: defaultTitle, onClick: testFunc })

    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
    // @ts-ignore
    wrapper.invoke('onClick')()

    expect(testFunc.calledOnce).toBeTruthy()
    done()
  })
})
