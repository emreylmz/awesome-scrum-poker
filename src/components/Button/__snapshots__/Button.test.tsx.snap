// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button UI should render correctly 1`] = `ShallowWrapper {}`;
